appengine/pond/duck/board/js/board.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Creates a view of all the ducks for a user.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Pond.Duck.Board');
+
+goog.require('BlocklyGames');
+goog.require('Pond.Datastore');
+goog.require('Pond.Duck.Board.soy');
+
+
+/**
+ * Initialize the board holding all the ducks for the current user.
+ */
+Pond.Duck.Board.init = function () {
+  //TODO: Have a loading screen before loading the ducks
```

Comment:
You could use the loading.gif, its used in a few places like in music.js where the div is included in the template and visibility is set to visible/hidden in JS
<img id="spinner" style="visibility: hidden;" src="common/loading.gif" height=15 width=15>

Reply:
<img id="spinner" style="visibility: hidden;" src="common/loading.gif" height=15 width=15>

Reply:
> <img id="spinner" style="visibility: hidden;" src="common/loading.gif" height=15 width=15>